java-checks/src/main/java/org/sonar/java/checks/PrintfCheck.java

```diff
@@ -341,8 +341,11 @@ private void checkTimeConversion(MethodInvocationTree mit, String param, Type ar
   }
 
   private void checkTimeTypeArgument(MethodInvocationTree mit, Type argType) {
-    if (!(argType.isNumerical() || argType.is("java.lang.Long") || argType.isSubtypeOf("java.util.Date") || argType.isSubtypeOf("java.util.Calendar"))) {
-      reportIssue(mit, "Time argument is expected (long, Long, Date or Calendar).");
+    if (!(argType.isNumerical()
+      || isOneOf(argType, "java.lang.Long", "java.time.LocalDate")
```

Comment:
I'd rename this method to `isTypeOfAny` and introduce `isSubtypeOfAny` for the following two lines. Also I tend to no use `!(a || b || c)` when it's too long, use `!a && !b && !c` instead.